src/main/java/com/github/maven_nar/NarCompileMojo.java

```diff
@@ -177,6 +177,22 @@ private void createLibrary(final Project antProject, final Library library)
       }
     }
 
+      if(getOS().equals( OS.WINDOWS ) && getArchitecture().equals("amd64"))
+      {
+
+          int noOfASMSources = getSourcesFor(getAssembler()).size();
+
+          if(noOfASMSources > 0)
+          {  // Assmbler files exist
+
```

Comment:
We have not included such a source file existance pre check on any of the other compiler types, not clear on need for this addition here.
With inclusion here getSourcesFor asm is happening twice, value is stored here but not re used.